src/array.jl

```diff
@@ -645,7 +645,7 @@ unique(A::CategoricalArray{T}) where {T} = _unique(T, A.refs, A.pool)
 Drop levels which do not appear in categorical array `A` (so that they will no longer be
 returned by [`levels`](@ref)).
 """
-droplevels!(A::CategoricalArray) = levels!(A, filter!(!ismissing, unique(A)))
+droplevels!(A::CategoricalArray) = levels!(A, intersect!(levels(A), unique(A)))
```

Comment:
`intersect!` appears only to support `IntSet`s on Julia 0.6, which is why the tests on 0.6 are failing.

Reply:
Thanks. Let's add a `VERSION` check then, anyway the performance difference is negligible.